ldms/src/ldmsd/ldmsd_config.c

```diff
@@ -311,11 +311,12 @@ void ldmsd_store___del(ldmsd_cfgobj_t obj)
  */
 int ldmsd_load_plugin(char* cfg_name, char *plugin_name, char *errstr, size_t errlen)
 {
-	struct ldmsd_plugin *api = load_plugin(plugin_name);
+	struct ldmsd_plugin *api;
+	if (!plugin_name || !cfg_name)
```

Comment:
I think it is good to only do the checking for the optional "plugin=" field in one place. But this change leaves the handling to the optional plugin= argument to the client rather than the server.

Older clients (e.g. maestro, ldmsd_controller), won't work with this approach of handling an optional "plugin=" on the client side, and it being required on the server side.

If we want backwards compatibility, the optional handling of "plugin=" should happen on the server (ldmsd) side. Clients should just pass through whatever arguments they have (i.e. always send "name=", only send "plugin=" when it is available).